crates/ra_ide_api/src/inlay_hints.rs

```diff
@@ -0,0 +1,189 @@
+use crate::{db::RootDatabase, FileId};
+use hir::{HirDisplay, Ty};
+use ra_syntax::ast::Pat;
+use ra_syntax::{
+    algo::visit::{visitor, Visitor},
+    ast::{self, PatKind, TypeAscriptionOwner},
+    AstNode, SmolStr, SourceFile, SyntaxNode, TextRange,
+};
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum InlayKind {
+    LetBinding,
```

Comment:
Let's try to keep `Inlay` more genereal than just types. So, `LetBindingType`, `ClosureParameterType`.

Reply:
Ok.